### DISCUSSION TITLE
Why can't the current collection add a one-to-many relationship to itself?

### DISCUSSION THREAD

**Discussion by cdwmhcc** [directus/directus#4704] [Archive]
I want to make a collection of `comments`, and there are many replies under each comment. It's similar to **Github's Discussions**.

Collection name: `comments`

**Field:**

`user_created`
`date_created`
`content`
`children`

I want to add a field with `children` many-to-one relationship, but when I added it, I found that I couldn't select the `comments` collection.

I know I can create two collections, one: `comments_parent` and the other: `comments_children`, and then add a many-to-one relationship to `comments_parent` in `comments_children`, or add a one-to-many relationship to `comments_children` in `comments_parent`. But I don't want to create two collections that have different collection names but the same other fields.

Maybe I got it wrong, please correct me.

🙏🙏🙏🙏🙏🙏🙏

**Comment by rijkvanzanten** [comment]
Funny that you bring it up, I just changed that in https://github.com/directus/directus/pull/4602.

FWIW, you can type in any name, even the name of the current collection and it'll work 👍🏻

**Reply by cdwmhcc** [reply]
The long-awaited feature! It also perfectly solves problems such as creating categories collections! Haha 👍🏻👍🏻👍🏻👍🏻👍🏻
